storage/src/mmr/mem.rs

```diff
@@ -99,12 +99,16 @@ impl<H: CHasher> Mmr<H> {
 
     /// Return the position of the oldest retained node in the MMR, not including the peaks which
     /// are always retained.
-    pub fn oldest_retained_pos(&self) -> u64 {
-        self.oldest_retained_pos
+    pub fn oldest_retained_pos(&self) -> Option<u64> {
+        if self.size() == 0 {
+            None
+        } else {
```

Comment:
nit: could drop the `else` and just do `return None;` above 🤷

Reply:
done